numpy/core/src/multiarray/item_selection.c

```diff
@@ -1577,6 +1577,10 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
         if (PyArray_SIZE(mps[0]) > 0) {
             *((npy_intp *)(PyArray_DATA(ret))) = 0;
         }
+
+        if (nd > 0 && check_and_adjust_axis(&axis, nd) < 0) {
```

Comment:
Hmmm, supporting `-1` for 0-D arrays seems actually wrong to me. `np.sort` errors here, maybe we should just change it or deprecate it? Unlike sort, `np.lexsort` does not support `axis=None` (ravel behaviour), although to be honest I am not sad about that :).

Reply:
Yeah noticed the axis=-1 issue, hence the nd > 0 check. Was not sure, if it was there for a reason, is there a specific reason we don't support ```axis=None``` for lexsort?

Reply:
I don't know if there is a reason. I suppose you would have to check the shapes match first and then ravel all inputs or so, instead of just raveling indiscriminately. So it may just be inconvenient to implement.
Or `sort` supports it mainly for some ancient back-compat reasons...

Reply:
Can we move this all the way up to line 1564?

Reply:
@seberg: We have some places where we allow `0` and `-1` as special-cases for 0d arrays. I don't particularly like it, but I think until we deprecate that we should do the same thing here.

As written, this also allows nonsense like `axis=1000` for 0d arrays, which isn't sane behavior,